engine/src/test/java/io/zeebe/engine/processing/streamprocessor/StreamProcessorReprocessingTest.java

```diff
@@ -601,4 +601,130 @@ public void processRecord(
+                              final TypedResponseWriter responseWriter,
+                              final TypedStreamWriter streamWriter,
+                              final Consumer<SideEffectProducer> sideEffect) {
+                            streamWriter.appendFollowUpEvent(
+                                record.getKey(),
+                                WorkflowInstanceIntent.ELEMENT_ACTIVATED,
+                                record.getValue());
+                          }
+                        })
+                    .onEvent(
+                        ValueType.WORKFLOW_INSTANCE,
+                        ELEMENT_ACTIVATED,
+                        new TypedRecordProcessor<UnifiedRecordValue>() {
+                          @Override
+                          public void processRecord(
+                              final long position,
+                              final TypedRecord<UnifiedRecordValue> record,
+                              final TypedResponseWriter responseWriter,
+                              final TypedStreamWriter streamWriter,
+                              final Consumer<SideEffectProducer> sideEffect) {}
+                        })
+                    .withListener(
+                        new StreamProcessorLifecycleAware() {
+                          @Override
+                          public void onRecovered(final ReadonlyProcessingContext context) {
+                            recoveredLatch.countDown();
+                          }
+                        }));
+
+    // then
+    recoveredLatch.await();
+
+    assertThat(streamProcessor.getLastProcessedPositionAsync().get()).isEqualTo(firstPosition);
+  }
+
+  @Test
+  public void shouldUpdateLastProcessedEventWhenSnapshot() throws Exception {
+    // given
+    final CountDownLatch onProcessedListenerLatch = new CountDownLatch(2);
+    streamProcessorRule.startTypedStreamProcessor(
+        (processors, context) ->
+            processors.onEvent(
+                ValueType.WORKFLOW_INSTANCE,
+                ELEMENT_ACTIVATING,
+                new TypedRecordProcessor<UnifiedRecordValue>() {
+                  @Override
+                  public void processRecord(
+                      final long position,
+                      final TypedRecord<UnifiedRecordValue> record,
+                      final TypedResponseWriter responseWriter,
+                      final TypedStreamWriter streamWriter,
+                      final Consumer<SideEffectProducer> sideEffect) {}
+                }),
+        (t) -> onProcessedListenerLatch.countDown());
+
+    streamProcessorRule.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING);
+    final var snapshotPosition =
+        streamProcessorRule.writeWorkflowInstanceEvent(
+            ELEMENT_ACTIVATING); // should be processed and included in the snapshot
+    onProcessedListenerLatch.await();
+    streamProcessorRule.snapshot();
+    streamProcessorRule.closeStreamProcessor();
```

Comment:
This part of the test is a bit hard to read. Please try to improve it. For example, by adding a new line :)